core/src/main/scala/spinal/core/AFix.scala

```diff
@@ -760,183 +759,213 @@ class AFix(val maxRaw: BigInt, val minRaw: BigInt, val exp: Int) extends MultiDa
+    this.getTag(classOf[TagAFixTruncated]).getOrElse(AFixTruncatedScope.get)
+  }
 
-  def round(exp: Int, aligned: Boolean): AFix = {
+  def round(exp: Int, aligned: Boolean = getTrunc.align): AFix = {
     val trunc = this.getTag(classOf[TagAFixTruncated])
 
     if (trunc.isDefined) {
-      this._round(trunc.get.rounding)
+      this._round(trunc.get.rounding, exp, aligned)
     } else {
-      roundHalfToInf(exp)
+      roundToInf(exp, aligned)
     }
   }
 
+  def fixTo(m: Int, exp: Int, roundType: RoundType): AFix = this.sat(m)._round(roundType, exp)
+  def fixTo(m: Int, exp: Int): AFix = this.fixTo(m, exp, getTrunc.rounding)
+  def fixTo(af: AFix, roundType: RoundType): AFix = this.sat(af)._round(roundType, af.exp)
+  def fixTo(af: AFix): AFix = this.fixTo(af, getTrunc.rounding)
+  def fixTo(Q: QFormat): AFix = {
+    val res = AFix(Q.width-Q.fraction exp, -Q.fraction exp, Q.signed)
+    res := this.fixTo(res)
+    res
+  }
+
   override def toString: String = s"${component.getPath() + "/" + this.getDisplayName()} : ${getClass.getSimpleName}[max=${maxRaw}, min=${minRaw}, exp=${exp}, bits=${raw.getWidth}]"
 
-  private def _round(roundType: RoundType): AFix = {
+  private def _round(roundType: RoundType, exp: Int = 0, align: Boolean = getTrunc.align): AFix = {
     roundType match {
-      case RoundType.FLOOR       => this.floor(0)
-      case RoundType.CEIL        => this.ceil(0)
-      case RoundType.FLOORTOZERO => this.floorToZero(0)
-      case RoundType.CEILTOINF   => this.ceilToInf(0)
-      case RoundType.ROUNDUP     => this.roundHalfUp(0)
-      case RoundType.ROUNDDOWN   => this.roundHalfDown(0)
-      case RoundType.ROUNDTOZERO => this.roundHalfToZero(0)
-      case RoundType.ROUNDTOINF  => this.roundHalfToInf(0)
-      case RoundType.ROUNDTOEVEN => this.roundHalfToEven(0)
-      case RoundType.ROUNDTOODD  => this.roundHalfToOdd(0)
-      case RoundType.SCRAP        => this.scrap(0)
+      case RoundType.FLOOR       => this.floor(exp)
+      case RoundType.CEIL        => this.ceil(exp, align)
+      case RoundType.FLOORTOZERO => this.floorToZero(exp)
+      case RoundType.CEILTOINF   => this.ceilToInf(exp, align)
+      case RoundType.ROUNDUP     => this.roundUp(exp, align)
+      case RoundType.ROUNDDOWN   => this.roundDown(exp, align)
+      case RoundType.ROUNDTOZERO => this.roundToZero(exp, align)
+      case RoundType.ROUNDTOINF  => this.roundToInf(exp, align)
+      case RoundType.ROUNDTOEVEN => this.roundToEven(exp, align)
+      case RoundType.ROUNDTOODD  => this.roundToOdd(exp, align)
+      case RoundType.SCRAP        => this.scrap(exp)
     }
   }
 
   def saturated(): AFix = this.truncated(saturation = true, overflow = false)
 
   def truncated(saturation: Boolean = false,
                 overflow  : Boolean = true,
-                rounding  : RoundType = RoundType.FLOOR) : AFix = {
+                rounding  : RoundType = RoundType.FLOOR,
+                align     : Boolean = true) : AFix = {
```

Comment:
What is the align argument ?

Reply:
It's a parameter used in all the rounding methods in Num[T], except for floor. 

It decides whether the rounding is aligned. If not aligned, the rounding result will prepend a carry bit at MSB, resulting in one bit longer than expected. Say rounding the last N bit from an M bit number without alignment will produce (M-N+1) bits instead of (M-N) bits (normal result).

Reply:
>  without alignment will produce (M-N+1) bits instead of (M-N) bits (normal result).

Shouldn't it be tracked via the maxRaw / minRaw ?
For instance see the "step-1" on : 
https://github.com/SpinalHDL/SpinalHDL/blob/703a281be9fc1e918e1f49431e3e22591b561c2c/core/src/main/scala/spinal/core/AFix.scala#L727

the "M-N+1" is to acount for rounding overflow ? (0xffff + 1) ?